python/pyarrow/tests/test_compute.py

```diff
@@ -376,6 +386,57 @@ def test_sum_chunked_array(arrow_type):
     assert pc.sum(arr, min_count=0).as_py() == 0
 
 
+@pytest.mark.parametrize('arrow_type', decimal_arrow_types)
+def test_sum_decimal_array(arrow_type):
```

Comment:
Thank you very much for your review. I have added the test cases you suggested into a separate function `test_decimal_compute`. However, I suspect that the assert `assert c.type==pa.decimal128(9,6)` may fail as it is adding 20 decimal places.

Reply:
I would like to ask if you mean `pc.multiply` as I can't see `multiple` in `pc`

Reply:
For the `assert` problem I suggested in the first comment, I believe it is the sole reason that causes the CI to fail.

Reply:
@deanm0000 Hi, would you mind if I ask should I leave the tests be failing because of the additional decimal place when multiplying?

Reply:
Should I open an issue for this?

Reply:
Could you work on one problem per issue/pull request?

Reply:
Hi, I was trying to address issue #44231 and I have only implemented test cases as @deanm0000 described in the issue. This is an additional thing that is required by him. Should I remove the relevant commits?

Reply:
Sorry for the delayed reply.

You're right, I meant `multiply` not `multiple`. On my bad decimal assert, that's my fault I was just making a quick comment. I certainly wouldn't leave it with my wrong statement. @kou I'm not sure if it should be considered the same issue or not. The issue I raised was incomplete tests so it's a judgement call on how in the weeds the tests should be and how many more basic tests there ought to be before it is testing for the sake of testing.

Reply:
How about using #44231 and this PR only for negative test values?
How about creating a new issue and PR for decimal related tests? (This may need to improve decimal related implementation? I haven't checked the failure yet.)

In general, small PRs are easy to review/merge.